addons/web_editor/static/lib/odoo-editor/src/utils/utils.js

```diff
@@ -1841,8 +1841,12 @@ export function setTagName(el, newTagName) {
     while (el.firstChild) {
         n.append(el.firstChild);
     }
+    const inLI = el.closest('li');
     if (el.tagName === 'LI') {
         el.append(n);
+    } else if (inLI && newTagName === 'p') {
+        inLI.removeChild(inLI.lastElementChild);
+        inLI.textContent = n.textContent;
```

Comment:
I do not understand this code. If we do `.textContent`, it will lose all the content that is not text. Also, why do we want to remove the last element child ? And what does it matter anyway since we are replacing it because of `.textContent =` anyway ? I'm very confused.

Reply:
Basically the extra padding of <p> leading to extra white space and when we make a text normal it wraps it in <p>, but in <LI> list items the text is not wrapped in <p> so we have to remove <p> and preserve the text content reassigning to <LI>. though i updated the code to remove el.

Reply:
Would it work if we replaced `inLI.textContent = n.textcontent` by `inLI.replaceChildren(...n.childNodes)` ? That way we do not lose the content that does not translate to text.

Also, I suggest to rename `inLI` to `closestLi`.

Reply:
okay it's working , updated the requested changes please check.
And also added a condition because it occurred to me that when a text is already normal and if you try to make it normal again it'll append \<p> so to avoid that i've added a condition.

Reply:
Thank you !